Wrong header is included

https://github.com/TileDB-Inc/TileDB/blob/51ba5af04b5153602d32c6258784689b2dd43c8c/tiledb/common/logger.h#L39
The wrong header is included here. The line should look like this:
```
#include <spdlog/fmt/fmt.h>
```
To ensure that the code always matches spdlog, the fmtlib of spdlog should be used and not something else.

Under Windows, the original line produces the following error message for me:
```
D:\TileDB\src\tiledb\common\logger.h(39,10): error C1083: File (Include) cannot be opened: “fmt/format.h”: No such file or directory [D:\TileDB\build\Debug\tiledb\common\baseline.vcxproj]
```
With the correction I can compile TileDB completely.

--- a/tiledb/common/logger.h
+++ b/tiledb/common/logger.h
@@ -36,7 +36,7 @@
 #ifndef TILEDB_LOGGER_H
 #define TILEDB_LOGGER_H
 
-#include <fmt/format.h>
+#include <spdlog/fmt/fmt.h>
 #include <atomic>
 #include <sstream>
 

